qp/pdf.py

```diff
@@ -124,31 +186,42 @@ def interpolate(self, vb=True):
 
         Notes
         -----
-        The `self.interpolator` object is a function, that is used by the `approximate` method.
+        The `self.interpolator` object is a function that is used by the `approximate` method.
         """
-        # First find the quantiles if none exist:
-        if self.quantiles is None:
-            self.quantiles = self.quantize()
+        if using == 'quantiles':
+            # First find the quantiles if none exist:
+            if self.quantiles is None:
+                self.quantiles = self.quantize()
+
+            self.difs = self.quantiles[1:]-self.quantiles[:-1]
+            self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
+            self.vals = (1.0/(len(self.quantiles)+1))/self.difs
 
-        self.difs = self.quantiles[1:]-self.quantiles[:-1]
-        self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
-        self.quantvals = (1.0/(len(self.quantiles)+1))/self.difs
+        if using == 'histogram':
+            # First find the histogram if none exists:
+            if self.histogram is None:
+                self.histogram = self.histogramize()
+
+            self.mids = (self.histogram[0][1:]+self.histogram[0][:-1])/2.
```

Comment:
I introduced `PDF.last` to store what parametrization was calculated most recently.  Now the interpolator uses that unless something else is specified, and it prints which parametrization that is.